### PR TITLE
Migrate from Jest to Vitest

### PR DESCRIPTION
**Pull request by nwalters512**
Jest has poor support for ESM, which is standing in the way of https://github.com/grafana/pyroscope-nodejs/pull/89. My hope is that this PR will unblock the ability to ship a functioning ESM distribution.

I took the liberty to remove some webpack dependencies and config files that, as far as I can tell, are unused. Let me know if you'd rather I do that in another PR.

### CODE REVIEW

File: package-lock.json
Comment (nwalters512) [inline package-lock.json:1]: AFAICT this file is no longer needed. Are there plans to merge https://github.com/grafana/pyroscope-nodejs/pull/90?

**Comment by bryanhuhta** [comment]
👋 Hey @nwalters512!

I seriously appreciate your batch of PRs here. These will go a long way to resolving a some repeated problems community members have reported.

I'm going to merge in the following order:

- https://github.com/grafana/pyroscope-nodejs/pull/90
- https://github.com/grafana/pyroscope-nodejs/pull/94
- https://github.com/grafana/pyroscope-nodejs/pull/95
- https://github.com/grafana/pyroscope-nodejs/pull/89 (hopefully this turns green after all the dust settles)

Then cut a new release.

**Comment by nwalters512** [comment]
@bryanhuhta 🎉 sounds great, I'm excited I'll be able to finally use Pyroscope in my app without weird hacks/workarounds. I'll be standing by to resolve merge conflicts or help resolve issues that might come up, particularly in #89.

**Comment by bryanhuhta** [comment]
@nwalters512 Looks like merging #90 caused conflicts in this set of PRs. Can you pull in changes from `main`?

**Comment by nwalters512** [comment]
@bryanhuhta done!

**Review by bryanhuhta — APPROVED** [review]

**Pull request merged**